engine/apps/user_management/middlewares.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import re
+
+import requests
+from django.http import HttpResponse
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework import status
+
+from apps.user_management.models.region import OrganizationMovedException
+from common.api_helpers.utils import create_engine_url
+
+logger = logging.getLogger(__name__)
+
+
+class OrganizationMovedMiddleware(MiddlewareMixin):
+    def process_exception(self, request, exception):
+        if isinstance(exception, OrganizationMovedException):
```

Comment:
As the plan stands currently handling this will continue to live in the engine, the gateway is going to focus on chat-ops back-ends & bots, main reason was we did not want to make the gateway aware of auth tokens and integrations.